### PR TITLE
trafficserver: init at 9.0.0

### LINKED ISSUE #111750: Traffic Server
**Project description**
Traffic Server is a Apache product.

**Metadata**

* homepage URL: http://trafficserver.apache.org/
* source URL: https://github.com/apache/trafficserver
* license: mit, bsd, gpl2+, etc.
* platforms: unix, linux, darwin, etc.

### LINKED ISSUE #113016: server/trafficserver: 9.0.0
**Project description**
Apache Traffic Server

**Metadata**

* homepage URL: https://trafficserver.apache.org
* source URL: https://downloads.apache.org/trafficserver/trafficserver-9.0.0.tar.bz2
* license: gpl2+
* platforms: unix

### PR DESCRIPTION
**Pull request by heroku-miraheze**
###### Motivation for this change
Traffic Server

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by asdf8dfafjk** [comment]
This cannot be merged since checkboxes have been filled incorrectly. (Sorry I had to channel my inner "heroku-miraheze"). 

On a serious note, you need to change the title to "init at 9.0.0" (at which point the last checkbox might make sense ...)

File: pkgs/servers/trafficserver/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  pname = "trafficserver";
+
+  # not using fetchFromGitHub as the git repo relies on submodules that are included in the tar file
+  src = fetchurl {
+    url = "https://www.apache.org/dyn/closer.cgi/trafficserver/trafficserver-${version}.tar.bz2";
+    sha256 = "sha256-1cffdff67711828df21d942ceff8acc798cdc50f5de1a7158edb5037fd4c6945";
+  };
+
+  nativeBuildInputs = [
+    gcc
```
Comment (SuperSandro2000) [inline pkgs/servers/trafficserver/default.nix:14]: I am pretty sure we don't need gcc or make here and binutils is probably also part of the stdenv.

File: pkgs/servers/trafficserver/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    make
+    binutils
+  ];
+
+configureFlags = [
+    "--with-experimental-plugins"
+];
```
Comment (SuperSandro2000) [inline pkgs/servers/trafficserver/default.nix:21]:
```suggestion
  configureFlags = [
    "--with-experimental-plugins"
  ];
```

File: pkgs/servers/trafficserver/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  enableParallelBuilding = true;
+
+  stripDebugList = [ "lib" "modules" "bin" ];
+
+  postInstall = ''
+    make install
+    mv $out /
```
Comment (SuperSandro2000) [inline pkgs/servers/trafficserver/default.nix:29]: What is this?

**Comment by heroku-miraheze** [comment]
Done.

**Comment by heroku-miraheze** [comment]
Merge it.

**Comment by 06kellyjac** [comment]
Hello @heroku-miraheze thanks for opening a PR

There are a couple things missing here

1. You need to add yourself to the `maintainer-list.nix` file before referring to `joaquinito2051` on line 39
2. You need to add a line to `all-packages.nix` before you can actually build the derivation
3. Your derivation is missing references such as `stdenv` and does not build so "Tested execution of all binary files" and "Tested compilation of all pkgs that depend on this change..." should not be checked off
4. There is no NixOS test here so please uncheck that
5. Your PR does not match `CONTRIBUTING.md` because:
  * Your file needs to be `nixpkgs-fmt`-ed
  * We put `pname` above `version`
  * We generally put smaller arrays on one line
  * You've now updated the PR name to the correct format but your commit needs to be the same

File: pkgs/servers/trafficserver/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Traffic Server";
+    homepage = "https://trafficserver.apache.org";
+    license = with licenses; [ asl20 gpl2Only ];
+    maintainers = with maintainers; [
+      joaquinito2051
```
Comment (SuperSandro2000) [inline pkgs/servers/trafficserver/default.nix:37]: Is this your maintainer name? Maybe you should change it to be similar to your GitHub name. If not please do not add other people as maintainers to packages.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
This is also missing an entry in all-packages.nix.

**Review by heroku-miraheze — COMMENTED** [review]
Traffic Server

**Comment by 06kellyjac** [comment]
I'm in the process of fixing this derivation, some more notes:

The sha is invalid
image

The actual file `https://www.apache.org/dyn/closer.cgi/trafficserver/trafficserver-9.0.0.tar.bz2` just fetches a webpage

image

image

---

Also I don't see any submodules in the repo so no reason not to use fetchurl

image

**Comment by sternenseemann** [comment]
> * Your file needs to be nixpkgs-fmt-ed
> * We put pname above version
> * We generally put smaller arrays on one line

These are all not reqirements for contributing as we don't have a formatting policy atm.

**Comment by 06kellyjac** [comment]
> > * Your file needs to be nixpkgs-fmt-ed
> > * We put pname above version
> > * We generally put smaller arrays on one line
> 
> These are all not reqirements for contributing as we don't have a formatting policy atm.

This might go a bit too far off track but:

I thought referring to `nixpkgs-fmt` was easier IMO than saying to indent the lines 19-21

Also it may not be in contributing but it's very common for changes to be requested even if it's not specifically defined in contributing.
I've been asked to change gemini links to https but that's not a `CONTRIBUTING.md` requirement

This fits `CONTRIBUTING.md` and the non-existant formatting requirements but no reviewer I'm familiar with would merge it

```nix
{ stdenv, lib }:




stdenv.mkDerivation
rec {
        version = "X";
pname = "X";

  # src stuff

  meta = with lib; {
homepage = "https://something";
  maintainers = with maintainers; [ someone ];
  license = licenses.mit;};
}
```

**Comment by 06kellyjac** [comment]
Here is a derivation that builds:

```nix
{ stdenv, lib, fetchFromGitHub, autoreconfHook, perl, openssl, pcre, zlib }:

stdenv.mkDerivation rec {
  pname = "trafficserver";
  version = "9.0.0";

  src = fetchFromGitHub {
    owner = "apache";
    repo = pname;
    rev = version;
    sha256 = "sha256-PmfgZGs0A5A21UpShPsBpzZULZPJYnbyeJLNuPgmeh8=";
  };

  nativeBuildInputs = [
    autoreconfHook
    perl
    openssl.dev
    pcre.dev
    zlib.dev
  ];

  configureFlags = [
    "--with-openssl=${openssl.dev}"
    "--with-pcre=${pcre.dev}"
    "--with-experimental-plugins"
  ];

  meta = with lib; {
    homepage = "https://trafficserver.apache.org";
    # Refers to a non-existant 10.0 and isn't very changelog-ie
    # changelog = "https://docs.trafficserver.apache.org/en/latest/release-notes/index.en.html";
    # Should exist according to previous releases (8 & 7) but hasn't been commited
    # changelog = "https://github.com/apache/trafficserver/blob/${version}/CHANGELOG-${version}";
    description = "A fast, scalable and extensible HTTP/1.1 and HTTP/2 compliant caching proxy server";
    longDescription = ''
      Traffic Server is a high-performance building block for cloud services.
      It's more than just a caching proxy server; it also has support for
      plugins to build large scale web applications.
    '';
    license = licenses.asl20;
    maintainers = with maintainers; [ joaquinito2051 ];
    platforms = platforms.all;
  };
}
```

<details>
  <summary>Result</summary>
  
result ⇒ /nix/store/dgdzy2hp8iw3h9k3i04j4plb0ylmy4m7-trafficserver-9.0.0
├── bin
│  ├── traffic_cache_tool
│  ├── traffic_crashlog
│  ├── traffic_ctl
│  ├── traffic_layout
│  ├── traffic_logcat
│  ├── traffic_logstats
│  ├── traffic_manager
│  ├── traffic_server
│  ├── traffic_via
│  ├── trafficserver
│  ├── tspush
│  └── tsxs
├── etc
│  └── trafficserver
│     ├── body_factory
│     │  └── default
│     │     ├── access#denied
│     │     ├── access#proxy_auth_required
│     │     ├── access#ssl_forbidden
│     │     ├── cache#not_in_cache
│     │     ├── cache#read_error
│     │     ├── connect#dns_failed
│     │     ├── connect#failed_connect
│     │     ├── connect#hangup
│     │     ├── default
│     │     ├── interception#no_host
│     │     ├── README
│     │     ├── redirect#moved_permanently
│     │     ├── redirect#moved_temporarily
│     │     ├── request#cycle_detected
│     │     ├── request#invalid_content_length
│     │     ├── request#no_content_length
│     │     ├── request#no_host
│     │     ├── request#scheme_unsupported
│     │     ├── request#syntax_error
│     │     ├── request#uri_len_too_long
│     │     ├── response#bad_response
│     │     ├── response#bad_version
│     │     ├── timeout#activity
│     │     ├── timeout#inactivity
│     │     ├── transcoding#unsupported
│     │     └── urlrouting#no_mapping
│     ├── cache.config
│     ├── hosting.config
│     ├── ip_allow.yaml
│     ├── logging.yaml
│     ├── parent.config
│     ├── plugin.config
│     ├── records.config
│     ├── remap.config
│     ├── sni.yaml
│     ├── socks.config
│     ├── splitdns.config
│     ├── ssl_multicert.config
│     ├── storage.config
│     ├── strategies.yaml
│     ├── trafficserver-release
│     └── volume.config
├── include
│  ├── ts
│  │  ├── apidefs.h
│  │  ├── experimental.h
│  │  ├── mgmtapi.h
│  │  ├── remap.h
│  │  └── ts.h
│  └── tscpp
│     ├── api
│     │  ├── Async.h
│     │  ├── AsyncHttpFetch.h
│     │  ├── AsyncTimer.h
│     │  ├── CaseInsensitiveStringComparator.h
│     │  ├── ClientRequest.h
│     │  ├── Continuation.h
│     │  ├── GlobalPlugin.h
│     │  ├── GzipDeflateTransformation.h
│     │  ├── GzipInflateTransformation.h
│     │  ├── Headers.h
│     │  ├── HttpMethod.h
│     │  ├── HttpStatus.h
│     │  ├── HttpVersion.h
│     │  ├── InterceptPlugin.h
│     │  ├── Logger.h
│     │  ├── noncopyable.h
│     │  ├── Plugin.h
│     │  ├── PluginInit.h
│     │  ├── RemapPlugin.h
│     │  ├── Request.h
│     │  ├── Response.h
│     │  ├── Stat.h
│     │  ├── Transaction.h
│     │  ├── TransactionPlugin.h
│     │  ├── TransformationPlugin.h
│     │  ├── Url.h
│     │  └── utils.h
│     └── util
│        ├── IntrusiveDList.h
│        ├── PostScript.h
│        └── TextView.h
├── lib
│  ├── libtscore.la
│  ├── libtscore.so ⇒ libtscore.so.9.0.0
│  ├── libtscore.so.9 ⇒ libtscore.so.9.0.0
│  ├── libtscore.so.9.0.0
│  ├── libtscppapi.la
│  ├── libtscppapi.so ⇒ libtscppapi.so.9.0.0
│  ├── libtscppapi.so.9 ⇒ libtscppapi.so.9.0.0
│  ├── libtscppapi.so.9.0.0
│  ├── libtscpputil.la
│  ├── libtscpputil.so ⇒ libtscpputil.so.9.0.0
│  ├── libtscpputil.so.9 ⇒ libtscpputil.so.9.0.0
│  ├── libtscpputil.so.9.0.0
│  ├── libtsmgmt.la
│  ├── libtsmgmt.so ⇒ libtsmgmt.so.9.0.0
│  ├── libtsmgmt.so.9 ⇒ libtsmgmt.so.9.0.0
│  ├── libtsmgmt.so.9.0.0
│  ├── perl5
│  │  ├── Apache
│  │  │  ├── TS
│  │  │  │  ├── AdminClient.pm
│  │  │  │  ├── Config
│  │  │  │  │  └── Records.pm
│  │  │  │  └── Config.pm
│  │  │  └── TS.pm
│  │  └── x86_64-linux-thread-multi
│  │     ├── auto
│  │     │  └── Apache
│  │     │     └── TS
│  │     └── perllocal.pod
│  ├── pkgconfig
│  │  └── trafficserver.pc
│  ├── plugin_init_fail.la
│  ├── plugin_init_fail.so
│  ├── plugin_instinit_fail.la
│  ├── plugin_instinit_fail.so
│  ├── plugin_missing_deleteinstance.la
│  ├── plugin_missing_deleteinstance.so
│  ├── plugin_missing_doremap.la
│  ├── plugin_missing_doremap.so
│  ├── plugin_missing_init.la
│  ├── plugin_missing_init.so
│  ├── plugin_missing_newinstance.la
│  ├── plugin_missing_newinstance.so
│  ├── plugin_required_cb.la
│  ├── plugin_required_cb.so
│  ├── plugin_testing_calls.la
│  ├── plugin_testing_calls.so
│  ├── plugin_v1.la
│  ├── plugin_v1.so
│  ├── plugin_v2.la
│  └── plugin_v2.so
├── libexec
│  └── trafficserver
│     ├── authproxy.la
│     ├── authproxy.so
│     ├── background_fetch.la
│     ├── background_fetch.so
│     ├── cache_promote.la
│     ├── cache_promote.so
│     ├── cache_range_requests.la
│     ├── cache_range_requests.so
│     ├── cachekey.la
│     ├── cachekey.so
│     ├── certifier.la
│     ├── certifier.so
│     ├── combo_handler.la
│     ├── combo_handler.so
│     ├── compress.la
│     ├── compress.so
│     ├── conf_remap.la
│     ├── conf_remap.so
│     ├── escalate.la
│     ├── escalate.so
│     ├── esi.la
│     ├── esi.so
│     ├── generator.la
│     ├── generator.so
│     ├── header_rewrite.la
│     ├── header_rewrite.so
│     ├── healthchecks.la
│     ├── healthchecks.so
│     ├── libloader.la
│     ├── libloader.so
│     ├── multiplexer.la
│     ├── multiplexer.so
│     ├── prefetch.la
│     ├── prefetch.so
│     ├── regex_remap.la
│     ├── regex_remap.so
│     ├── regex_revalidate.la
│     ├── regex_revalidate.so
│     ├── remap_purge.la
│     ├── remap_purge.so
│     ├── s3_auth.la
│     ├── s3_auth.so
│     ├── server_push_preload.la
│     ├── server_push_preload.so
│     ├── stats_over_http.la
│     ├── stats_over_http.so
│     ├── tcpinfo.la
│     ├── tcpinfo.so
│     ├── xdebug.la
│     └── xdebug.so
├── share
│  ├── man
│  │  └── man3
│  │     ├── Apache::TS.3.gz
│  │     ├── Apache::TS::AdminClient.3.gz
│  │     └── Apache::TS::Config::Records.3.gz
│  └── trafficserver
└── var
   ├── log
   │  └── trafficserver
   └── trafficserver

</details>

File: pkgs/servers/trafficserver/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  postInstall = ''
+    make install
+    mv $out /
+  '';
+
+  meta = with lib; {
+    description = "Traffic Server";
+    homepage = "https://trafficserver.apache.org";
+    license = with licenses; [ asl20 gpl2Only ];
```
Comment (dotlambda) [inline pkgs/servers/trafficserver/default.nix:35]:
```suggestion
    license = licenses.asl20;
```
I don't see GPL in https://github.com/apache/trafficserver/blob/9.0.0/LICENSE.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Pull request closed without merging**